app/backend/src/couchers/helpers/geoip.py

```
@@ -10,31 +10,38 @@
 
 def geoip_approximate_location(ip_address: str) -> str | None:
     if config["GEOLITE2_CITY_MMDB_FILE_LOCATION"] == "":
-        return
+        return None
     if ip_address is None:
-        return
+        return None
     try:
         with geoip2.database.Reader(config["GEOLITE2_CITY_MMDB_FILE_LOCATION"]) as reader:
             response = reader.city(ip_address)
             city = response.city.name
             country = response.country.name
             return f"{city}, {country}" if city else f"{country}"
     except AddressNotFoundError:
-        pass
+        return None
     except Exception as e:
         logger.error(f"GeoIP failed for {ip_address=}")
+        return None
 
 
 def geoip_asn(ip_address: str) -> tuple[int, str, str] | None:
     if config["GEOLITE2_ASN_MMDB_FILE_LOCATION"] == "":
-        return
+        return None
     if ip_address is None:
-        return
+        return None
     try:
         with geoip2.database.Reader(config["GEOLITE2_ASN_MMDB_FILE_LOCATION"]) as reader:
             response = reader.asn(ip_address)
-            return response.autonomous_system_number, response.autonomous_system_organization, response.network
+            asn_number = response.autonomous_system_number
+            asn_org = response.autonomous_system_organization
+            network_str = str(response.network)
+            if asn_number is not None and asn_org is not None:
+                return asn_number, asn_org, network_str
+            return None
```

Why change the logic?

---

I now changed it to raise an exception if the values are None. The calling code assumes that the fields are not None, so we should throw an error here.